app/simulation_svc.py

```diff
@@ -70,7 +70,7 @@ async def _generate_simulated_response(self, link_id):
         words = []
         for x in range(randint(1, 100)):
             words.append(self.generate_name(size=randint(2, 10)))
-        await ResultGenerator(ability.parsers).generate(words)
+        await ResultGenerator(getattr(ability, 'parser', [])).generate(words)
```

Comment:
I wonder if it might be better to do the checks in the ResultGenerator's `__init__` function, like `self.parser = parser if parser else []`. But this seems fine to me too